test/test_roscpp/test/src/params.cpp

```diff
@@ -528,6 +528,38 @@ TEST(Params, mapBoolParam)
   ASSERT_TRUE(std::equal(map_b.begin(), map_b.end(), map_b2.begin()));
 }
 
+TEST(Params, paramTemplateFunction)
+{
+  EXPECT_EQ( param::param<std::string>( "string", "" ), "test" );
+  EXPECT_EQ( param::param<std::string>( "gnirts", "test" ), "test" );
+
+  EXPECT_TRUE( param::param<int>( "int", 0 ) == 10 );
+  EXPECT_TRUE( param::param<int>( "tni", 10 ) == 10 );
+
+  EXPECT_DOUBLE_EQ( param::param<double>( "double", 0.0 ), 10.5 );
+  EXPECT_DOUBLE_EQ( param::param<double>( "elbuod", 10.5 ), 10.5 );
+
+  EXPECT_FALSE( param::param<bool>( "bool", true ) );
+  EXPECT_TRUE( param::param<bool>( "loob", true ) );
+}
+
+TEST(Params, paramNodeHandleTemplateFunction)
+{
+  NodeHandle nh;
+
+  EXPECT_EQ( nh.param<std::string>( "string", "" ), "test" );
+  EXPECT_EQ( nh.param<std::string>( "gnirts", "test" ), "test" );
+
+  EXPECT_TRUE( nh.param<int>( "int", 0 ) == 10 );
+  EXPECT_TRUE( nh.param<int>( "tni", 10 ) == 10 );
+
+  EXPECT_DOUBLE_EQ( nh.param<double>( "double", 0.0 ), 10.5 );
+  EXPECT_DOUBLE_EQ( nh.param<double>( "elbuod", 10.5 ), 10.5 );
+
+  EXPECT_FALSE( nh.param<bool>( "bool", true ) );
```

Comment:
I'd replace this with EXPECT_EQ too. true/false are values returned by `nh.param<>()`, it's not like it returns a binary answer (as opposed to methods like `is_open()` or `is_valid()`), there's a slight semantic difference IMO. @dirk-thomas @tfoote @wjwwood thoughts?